### ISSUE TITLE
"found blacklisted journal entry[...] fsck_errors_not_fixed" [74a993e2619b]

### ISSUE THREAD

**Issue by marcin-github** [koverstreet/bcachefs#486]
Hi,
I broke fs. I have fs in good state then I run `fsck -y` but after hour (two?) I hit ctrl+c . Now I have:

```
[   44.379209] bcachefs (dm-8): journal read done on device dm-8, ret 0
[   44.379302] bcachefs (dm-8): journal read done, replaying entries 93233-93233
[   44.379353] found blacklisted journal entry 93233, exiting
[   44.379358] bcachefs (dm-8): Unable to continue, halting
[   44.379370] bcachefs (dm-8): Error in recovery: cannot allocate memory (fsck_errors_not_fixed)
[   44.379375] bcachefs (dm-8): error starting filesystem: fsck_errors_not_fixed
[   44.379379] bcachefs (dm-8): shutting down
[   44.383382] bcachefs (dm-8): shutdown complete
```

fsck in kernel can't repair this fs. What to do now?

//edit:
show-super:
```
External UUID:                              765118bc-a2e8-419b-8100-97cdf1a0c6ee
Internal UUID:                              47f89b5e-fc28-41d1-acc4-2fc4db82a4bc
Device index:                               0
Label:                                      media1
Version:                                    unwritten_extents
Oldest version on disk:                     unwritten_extents
Created:                                    Wed Nov 23 21:42:25 2022
Sequence number:                            93
Superblock size:                            3904
Clean:                                      0
Devices:                                    1
Sections:                                   members,replicas_v0,clean,journal_seq_blacklist,journal_v2,counters
Features:                                   lz4,gzip,zstd,journal_seq_blacklist_v3,reflink,new_siphash,inline_data,new_extent_overwrite,incompressible,btree_ptr_v2,extents_
above_btree_updates,btree_updates_journalled,new_varint,journal_no_flush,alloc_v2,extents_across_btree_nodes
Compat features:                            alloc_info,alloc_metadata,extents_above_btree_updates_done,bformat_overflow_done

Options:                                                                                                                                                            [1/1549]
  block_size:                               4.00 KiB
  btree_node_size:                          256 KiB
  errors:                                   continue [ro] panic
  metadata_replicas:                        1
  data_replicas:                            1
  metadata_replicas_required:               1
  data_replicas_required:                   1
  encoded_extent_max:                       1.00 MiB
  metadata_checksum:                        none [crc32c] crc64 xxhash
  data_checksum:                            none [crc32c] crc64 xxhash
  compression:                              none [lz4] gzip zstd
  background_compression:                   [none] lz4 gzip zstd
  str_hash:                                 crc32c crc64 [siphash]
  metadata_target:                          none
  foreground_target:                        none
  background_target:                        none
  promote_target:                           none
  erasure_code:                             0
  inodes_32bit:                             1
  shard_inode_numbers:                      1
  inodes_use_key_cache:                     1
  gc_reserve_percent:                       5
  gc_reserve_bytes:                         0 B
  root_reserve_percent:                     0
  wide_macs:                                1
  acl:                                      1
  usrquota:                                 0
  grpquota:                                 0
  prjquota:                                 0
  journal_flush_delay:                      1000
  journal_flush_disabled:                   0
  journal_reclaim_delay:                    100
  nocow:                                    0

members (size 64):
  Device:                                   0
    UUID:                                   99df5dc4-a049-40a8-b000-d8af4a1b84f2
    Size:                                   3.32 TiB
    Bucket size:                            512 KiB
    First bucket:                           0
    Buckets:                                6963200
    Last mount:                             Thu Dec  8 19:08:13 2022
    State:                                  rw
    Label:                                  (none)
    Data allowed:                           journal,btree,user
    Has data:                               journal,btree,user
    Discard:                                0
    Freespace initialized:                  1
```

**Comment by bhzhu203** [comment]
"cannot allocate memory"

 you need add more physical memory to the machine.Because the fsck need more memory

**Comment by marcin-github** [comment]
Now I have 16GB, how much I need?

**Comment by bhzhu203** [comment]
> Now I have 16GB, how much I need?

You can keep wacthing the fsck tool memory in "top" when doing fsck. 

Last time I had done a  fsck   on a 8T single HDD. It use more than 30GB memory...  I am  so astonished. 

It should be improved @koverstreet 


By the way , you could try to use zram when the memory is limited.


modprobe zram
echo 8G > /sys/block/zram0/disksize    // half of pysical memory 
mkswap /dev/zram0
swapon --priority 100 /dev/zram0


```
echo "
vm.extfrag_threshold=0
vm.swappiness=100
vm.page-cluster=0

vm.min_free_kbytes=220000
vm.dirty_background_ratio=5
vm.dirty_ratio = 10
vm.dirty_expire_centisecs = 1500
vm.dirty_writeback_centisecs = 300
vm.vfs_cache_pressure=150
" >> /etc/sysctl.conf 

sysctl -p
``````

**Comment by koverstreet** [comment]
It's not a memory issue

**Comment by marcin-github** [comment]
Dnia 13 grudnia 2022 21:04:34 CET, koverstreet ***@***.***> napisał/a:
>Why can't fsck repair? Show me the fsck output
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/koverstreet/bcachefs/issues/486#issuecomment-1349616987
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
Dmesg was from mount -o fsck,fix_errors, i forget to write it. 
-- Wysłane za pomocą K-9 Mail.

**Comment by koverstreet** [comment]
Fixed by ca2b096ed0
